be/src/vec/columns/column_dictionary.h

```diff
@@ -106,6 +107,7 @@ class ColumnDictionary final : public COWHelper<IColumn, ColumnDictionary<T>> {
     void clear() override {
         _codes.clear();
         _dict_code_converted = false;
+        _dict.clear_hash_values();
```

Comment:
`hash_values` only needs to be calculated once.
If `clear_hash_values` is called here, the hash of the dictionary values needs to be calculated once for each call to  `SegmentIterator::next_batch`.

Reply:
```convert_dict_codes_if_necessary``` may change code order, so we need to reset every time.
Of course, we can solve it but may introduce additional complexity.
I didn't see the significant performance improvement in tpcds-q60, this may need further research.
Currently I prefer to keep status quo.